### PR TITLE
get partitioning measures: 229

### PR DESCRIPTION
**Pull request by TebaleloS**
Release notes:

* Endpoint functionality to get partitioning measures by partitioning ID.
* Defined SQL function to get the measures by ID.
* Defined and implemented the class to call the DB function.
* Implemented the endpoint functionality all the way from the repository to the endpoint.
* Added integration and unit tests.

### CODE REVIEW

**Comment by TebaleloS** [comment]
Notes:

* Endpoint functionality to get partitioning measures by partitioning ID.

File: server/src/main/scala/za/co/absa/atum/server/api/http/BaseEndpoints.scala [hunk 1]
```diff
@@ -71,7 +71,8 @@ trait BaseEndpoints {
       oneOf[ErrorResponse](
         badRequestOneOfVariant,
         generalErrorOneOfVariant,
-        internalServerErrorOneOfVariant
+        internalServerErrorOneOfVariant,
+        notFoundErrorOneOfVariant
```
Comment (salamonpavel) [inline server/src/main/scala/za/co/absa/atum/server/api/http/BaseEndpoints.scala:75]: Not here in the base endpoint, add this to the GET endpoint for partitioning measures. The base endpoint should provide only things common to every single endpoint.

File: server/src/test/scala/za/co/absa/atum/server/api/repository/PartitioningRepositoryUnitTests.scala [hunk 2]
```diff
@@ -240,6 +253,28 @@ object PartitioningRepositoryUnitTests extends ZIOSpecDefault with TestData {
...
+          assertZIO(PartitioningRepository.getPartitioningMeasuresById(2L).exit)(
+            failsWithA[NotFoundDatabaseError]
+          )
+        },
+        test("Returns expected NotFoundDatabaseError") {
+          assertZIO(PartitioningRepository.getPartitioningMeasuresById(3L).exit)(
+            failsWithA[NotFoundDatabaseError]
+          )
+        },
+        test("Returns expected DatabaseError") {
```
Comment (salamonpavel) [inline server/src/test/scala/za/co/absa/atum/server/api/repository/PartitioningRepositoryUnitTests.scala:273]: GeneralDatabaseError

File: server/src/main/scala/za/co/absa/atum/server/api/http/BaseEndpoints.scala (hunk 1 above)
Comment (TebaleloS) [reply server/src/main/scala/za/co/absa/atum/server/api/http/BaseEndpoints.scala:75]: I actually didn't check it. Thank you for the alert, I will remove it.

**Review by salamonpavel — APPROVED** [review]

**Review by benedeki — APPROVED** [review]
- [x] code reviewed
- [x] pulled
- [ ] built
- [ ] run

**Pull request merged**